runtime/src/main/cpp/Memory.cpp

```diff
@@ -1216,8 +1206,9 @@ ContainerHeader* AllocContainer(MemoryState* state, size_t size) {
   // We recycle elements of finalizer queue for new allocations, to avoid trashing memory manager.
   ContainerHeader *container = state->finalizerQueue, *previous = nullptr;
   while (container != nullptr) {
-    // TODO: shall it be >= or ==?
-    if (container->hasContainerSize() && container->containerSize() == size) {
+    // TODO: shall it be == instead?
+    if (container->hasContainerSize() &&
+        container->containerSize() >= size && container->containerSize() <= 2 * size) {
```

Comment:
`container->containerSize() <= 2 * size` is too permissive. Can it be more like `container->containerSize() <= size + 8`?